pkg/metrics/certificates.go

```diff
@@ -61,6 +63,17 @@ func (m *Metrics) updateCertificateExpiry(ctx context.Context, key string, crt *
 		"namespace": crt.Namespace}).Set(expiryTime)
 }
 
+// updateCertificateRenewBefore updates the renew before duration of a certificate
+func (m *Metrics) updateCertificateRenewBefore(crt *cmapi.Certificate) {
+	renewBefore := crt.Spec.RenewBefore
```

Comment:
Should we report on the adjusted duration used by _cert-manager_ when this value is greater than the requested life span of the certificate (that is, 1/3 of that life span)?

Reply:
I was planning on relying on prometheus rules for this. What sort of metric would you expect for reporting that. A gauge that is 1 or 0?

Reply:
I was suggesting reporting not what the _Certificate_'s spec requests, but rather what _cert-manager_ actually winds up using, per [the `certificates.RenewalTime` function](https://sourcegraph.com/github.com/jetstack/cert-manager@168a4b35a24452753028dfc67277f9a3e35e32b5/-/blob/pkg/controller/certificates/util.go?L289-L322).

Reply:
Ah, I don't think I knew that existed. Looking closer, certificates have a `Status.RenewalTime`. Should I just report that instead? Then you prometheus alert could be `RenewalTime < time()`

Reply:
Using the "status.renewalTime" field makes sense to me, but you may need to accommodate _Certificate_ objects that lack that field.

Reply:
Should it be seperate metrics? In what case would a Certificate object not have a status.renewalTime?

Reply:
I haven't studied the _cert-manager_ code to audit this, but it's possible that newly created _Certificates_ won't have their "status" field populated at all (apart from the zero values). Not until the controller reconciles the new object will it write back to its "status" subresource.

Reply:
Ok, that would be similar to the expiration time metric then. Where it sets it to 0 if the status is nil.

I think that would achieve the original goal of this PR. Which is to be able to specify an alert when a certificate should have renewed itself but has not.

So a prometheus rule could be

```
certmanager_certificate_renew_timestamp < time()
```

This would be true anytime the renew_timestamp was passed.

@jakexks any thoughts?

Reply:
Just pushed this up. Seems to be working well on my test cluster